csharp/src/Client/SchemaConverter.cs

```diff
@@ -70,8 +70,17 @@ public static DataTable ConvertArrowSchema(Schema schema, AdbcStatement adbcStat
                     f.HasMetadata
                 )
                 {
-                    row[SchemaTableColumn.NumericPrecision] = Convert.ToInt32(f.Metadata["precision"]);
-                    row[SchemaTableColumn.NumericScale] = Convert.ToInt32(f.Metadata["scale"]);
+                    if (f.Metadata.TryGetValue("precision", out string precisionKey))
```

Comment:
consider for slightly greater robustness

`    if (f.Metadata.TryGetValue("precision", out string precisionKey) && int32.TryParse(precisionKey, out int precision))`